### DISCUSSION TITLE
Cannot read cookies from server page component?

### DISCUSSION THREAD

**Discussion by oelbaga** [vercel/next.js#42521] [Help]
I can read and write cookies that are set in the /pages/api folder from the page component (that lives in the /app folder) only when I set the page component to a client component with 'use client'

But when I leave the page component to its default as a server component and call the cookie from the /pages/api folder the cookie isn't read.

Thoughts?

**Comment by soylemezali42** [comment]
Hi @oelbaga  did you solve that problem. I have a similiar issue too. I couldn't read the cookie in server component.
